src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -187,169 +190,115 @@ def interpolate_h5_file(source_file_path,
-                        choices=quadrature_dict.keys(),
-                        required=True,
-                        help=("The quadrature of the target grid. "
-                              "Can be different for each dimension"))
-
-    parser.add_argument(
-        "--start-time",
-        type=float,
-        default=-np.inf,
-        help=("Disregard all observations with value before this point"))
-
-    parser.add_argument(
-        "--stop-time",
-        type=float,
-        default=np.inf,
-        help=("Disregard all observations with value after this point"))
-
-    parser.add_argument(
-        "--stride",
-        type=int,
-        default=1,
-        help=("Stride through observations with this step size."))
-
-    parser.add_argument(
-        "-j",
-        "--jobs",
-        type=int,
-        default=None,
-        help=("The maximum number of processes to be started. "
-              "A process is spawned for each source file up to this number."))
-
-    parser.add_argument('-v',
-                        '--verbose',
-                        action='count',
-                        default=0,
-                        help="Verbosity (-v, -vv, ...)")
-
-    return parser.parse_args(sys_args)
-
-
-if __name__ == "__main__":
-
-    parsed_args = parse_args(sys.argv[1:])
-
-    logging.basicConfig(level=logging.WARNING - parsed_args.verbose * 10)
-    source_files = glob.glob(parsed_args.source_file_prefix + "[0-9]*.h5")
+@click.command(help=interpolate_h5_file.__doc__)
+@click.option(
+    "--source-file-prefix",
+    required=True,
+    help=("The prefix for the .h5 source files. All files starting with the "
+          "prefix followed by a number will be interpolated."))
+@click.option("--source-subfile-name",
+              required=True,
+              help=("The name of the volume data subfile within the "
+                    "source files in which the data is contained"))
+@click.option(
+    "--target-file-prefix",
+    default=None,
+    help=(
+        "The prefix for the target files where the interpolated data is "
+        "written. When no target file is specified, the interpolated data is "
+        "written to the corresponding source file in a new volume data "
+        "subfile."))
+@click.option("--target-subfile-name",
+              required=True,
+              help=("The name of the volume data subfile within the target "
+                    "files where the data will be written."))
+@click.option("--tensor-component",
+              "-t",
+              multiple=True,
+              help=("The names of the tensors that are to be interpolated. "
+                    "Accepts regular expression. "
+                    "If none are specified, all tensors are interpolated."))
+@click.option(
+    "--target-extents",
+    callback=(lambda ctx, param, value: list(map(int, value.split(',')))
+              if value else []),
+    required=True,
+    help=("The extents of the target grid. "
```

Comment:
It should be given as a comma-separated list without spaces e.g. ...

The error message with the old argparse style `3 4 5` is not very helpful because it thinks you are giving more command line options